native/packages/react-native-bpk-component-banner-alert/src/BpkBannerAlert.android.js

```diff
@@ -0,0 +1,255 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  StyleSheet,
+  TouchableNativeFeedback,
+  TouchableWithoutFeedback,
+  View,
+  ViewPropTypes,
+} from 'react-native';
+
+import {
+  borderRadiusSm,
+  colorBlue500,
+  colorGray50,
+  colorGray500,
+  colorGreen500,
+  colorRed500,
+  colorYellow500,
+  spacingBase,
+  spacingLg,
+  spacingMd,
+  spacingSm,
+  spacingXxl,
+} from 'bpk-tokens/tokens/base.react.native';
+
+import BpkAnimateHeight from 'react-native-bpk-component-animate-height';
+import BpkIcon from 'react-native-bpk-component-icon';
+import BpkText from 'react-native-bpk-component-text';
+
+import AnimateAndFade from './AnimateAndFade';
+import {
+  dismissablePropType,
+  toggleExpandedButtonLabelPropType,
+  dismissableLabelPropType,
+} from './customPropTypes';
+
+export const ALERT_TYPES = {
+  SUCCESS: 'success',
+  WARN: 'warn',
+  ERROR: 'error',
+  NEUTRAL: 'neutral',
+};
+
+const styles = StyleSheet.create({
+  mainContainer: {
+    paddingVertical: spacingBase,
+    paddingHorizontal: spacingLg,
+    backgroundColor: colorGray50,
+    minHeight: spacingXxl + spacingBase,
+    flex: 1,
+  },
+  dismissableContainer: {
+    paddingHorizontal: spacingLg - spacingMd,
+    backgroundColor: colorGray50,
+    justifyContent: 'center',
+  },
+  childrenContainer: {
+    paddingVertical: spacingSm,
+  },
+  flexRow: {
+    display: 'flex',
+    flexDirection: 'row',
+  },
+  text: {
+    flex: 1,
+  },
+  icon: {
+    paddingRight: spacingSm,
+    paddingTop: spacingSm - spacingSm / spacingSm,
+  },
+  dismissLabel: {
+    color: colorBlue500,
+  },
+  dismissButtonContainer: {
+    padding: spacingMd,
+    borderRadius: borderRadiusSm,
+  },
+  dismissButtonContainerRadius: {
+    borderRadius: borderRadiusSm,
+  },
+});
+
+const ALERT_TYPE_STYLES = {
+  [ALERT_TYPES.SUCCESS]: {
+    iconSource: 'tick-circle',
+    iconStyle: {
+      color: colorGreen500,
+    },
+  },
+  [ALERT_TYPES.WARN]: {
+    iconSource: 'information-circle',
+    iconStyle: {
+      color: colorYellow500,
+    },
+  },
+  [ALERT_TYPES.ERROR]: {
+    iconSource: 'information-circle',
+    iconStyle: {
+      color: colorRed500,
+    },
+  },
+  [ALERT_TYPES.NEUTRAL]: {
+    iconSource: 'information-circle',
+    iconStyle: {
+      color: colorGray500,
+    },
+  },
+};
+
+const BpkBannerAlert = props => {
+  const {
+    type,
+    message,
+    onDismiss,
+    onToggleExpanded,
+    dismissable,
+    expanded,
+    dismissButtonLabel,
+    toggleExpandedButtonLabel,
+    show,
+    style: userStyle,
+    animateOnEnter,
+    animateOnLeave,
+    children,
+    ...rest
+  } = props;
+  const expandable = children !== null;
+  const { iconSource, iconStyle } = ALERT_TYPE_STYLES[type] || {};
+
+  const dismissableButton = () => {
```

Comment:
It's probably not great to create this component inside render every time. I would suggest moving it out

Reply:
I think it also doesn't need to be a function - the only place it's called is on line 197, that looks like this:

``` {dismissable && dismissableButton()}```

Therefore you don't need to do `if (!dismissable) { return null; }` as that's already been checked. 

I recommend changing it to be a variable, same as you have for `expandableContent` and `expandableIcon`.

Reply:
Fair, will do